File: point.c
/*
 * point.c
 *
 *  Created on: 18 Sep 2018
 *      Author: Diogo
 */

#include <stdlib.h>
#include <math.h>
#include "point.h"

typedef struct _point{
	float x, y;
}infoPoint, *apPoint;

point newPoint(float x, float y){
	apPoint nPoint = (apPoint)malloc(sizeof(infoPoint));
	nPoint->x = x;
	nPoint->y = y;
	return (point)nPoint;
}

void setVals(point a, float x, float y){

	((apPoint)a)->x = x;
	((apPoint)a)->y = y;

}

float distPoints(point a, point b){

	float xDif = ((apPoint)a)->x - ((apPoint)b)->x;
	float yDif = ((apPoint)a)->y - ((apPoint)b)->y;

	return sqrt(xDif*xDif + yDif*yDif);
}

float getX(point a){
	return ((apPoint)a)->x;
}

float getY(point a){
	return ((apPoint)a)->y;
}

File: T2Q1/Point.java
public class Point {
	
	private double x, y;
	
	//Default contructor. X and Y = 0
	public Point() {
		x = 0;
		y = 0;
	}
	
	public Point(double _x, double _y) {
		x = _x;
		y = _y;
	}
	
	public double getX() {
		return x;
	}
	
	public double getY() {
		return y;
	}
	
	public double distP(Point _a) {
		double xDif, yDif;
		xDif = x - _a.getX();
		yDif = y - _a.getY();
		
		return Math.sqrt(xDif*xDif + yDif*yDif);
	}
}
